pkg/controller/actions/deploy/action_deploy_test.go

```diff
@@ -239,3 +250,133 @@ func TestDeployNotOwnedCreate(t *testing.T) {
+	utilruntime.Must(componentsv1.AddToScheme(s))
+	utilruntime.Must(rbacv1.AddToScheme(s))
+
+	projectDir, err := envtestutil.FindProjectRoot()
+	g.Expect(err).NotTo(HaveOccurred())
+
+	envTest := &envtest.Environment{
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Scheme: s,
+			Paths: []string{
+				filepath.Join(projectDir, "config", "crd", "bases"),
+			},
+			ErrorIfPathMissing: true,
+			CleanUpAfterUse:    false,
+		},
+	}
+
+	t.Cleanup(func() {
+		_ = envTest.Stop()
+	})
+
+	cfg, err := envTest.Start()
+	g.Expect(err).NotTo(HaveOccurred())
+
+	envTestClient, err := client.New(cfg, client.Options{Scheme: s})
+	g.Expect(err).NotTo(HaveOccurred())
+
+	cli, err := odhCli.NewFromConfig(cfg, envTestClient)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	t.Run("aggregation", func(t *testing.T) {
+		ctx := context.Background()
+		name := xid.New().String()
+
+		deployClusterRoles(t, ctx, cli, rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Rules: []rbacv1.PolicyRule{{
+				Verbs:     []string{"*"},
+				Resources: []string{"*"},
+				APIGroups: []string{"*"},
+			}},
+			AggregationRule: &rbacv1.AggregationRule{
+				ClusterRoleSelectors: []metav1.LabelSelector{{
+					MatchLabels: map[string]string{"foo": "bar"},
+				}},
+			},
+		})
+
+		out := rbacv1.ClusterRole{}
+		err = cli.Get(ctx, client.ObjectKey{Name: name}, &out)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		g.Expect(out).To(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Rules": BeEmpty(),
+		}))
+	})
+
+	t.Run("no aggregation", func(t *testing.T) {
+		ctx := context.Background()
+		name := xid.New().String()
+
+		deployClusterRoles(t, ctx, cli, rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Rules: []rbacv1.PolicyRule{{
+				Verbs:     []string{"*"},
+				Resources: []string{"*"},
+				APIGroups: []string{"*"},
+			}},
+		})
+
+		out := rbacv1.ClusterRole{}
+		err = cli.Get(ctx, client.ObjectKey{Name: name}, &out)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		g.Expect(out).To(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Rules": HaveLen(1),
+		}))
+	})
+}
+
+func deployClusterRoles(t *testing.T, ctx context.Context, cli *odhCli.Client, roles ...rbacv1.ClusterRole) {
+	t.Helper()
+
+	g := NewWithT(t)
+
+	rr := types.ReconciliationRequest{
+		Client: cli,
+		DSCI: &dsciv1.DSCInitialization{Spec: dsciv1.DSCInitializationSpec{
+			ApplicationsNamespace: xid.New().String(),
+		}},
+		DSC: &dscv1.DataScienceCluster{},
+		Instance: &componentsv1.Dashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Generation: 1,
+				UID:        apimachinery.UID(xid.New().String()),
+			},
+		},
+		Release: cluster.Release{
+			Name: cluster.OpenDataHub,
+			Version: version.OperatorVersion{Version: semver.Version{
+				Major: 1, Minor: 2, Patch: 3,
+			}}},
+	}
+
+	for i := range roles {
+		g.Expect(
+			rr.AddResources(roles[i].DeepCopy()),
+		).ShouldNot(
+			HaveOccurred(),
```

Comment:
means "error" should not have occurred? ;)

Reply:
yep :)